src/sdk/namespace_fs.js

```diff
@@ -602,27 +606,33 @@ class NamespaceFS {
                 if (drain_promise) {
                     await drain_promise;
                     drain_promise = null;
+                    object_sdk.throw_if_aborted();
                 }
 
                 // write the data out to response
                 buffer_pool_cleanup = null; // cleanup is now in the socket responsibility
                 const write_ok = res.write(data, null, callback);
                 if (!write_ok) {
-                    drain_promise = stream_utils.wait_drain(res);
+                    drain_promise = stream_utils.wait_drain(res, object_sdk.abort_controller.signal);
                 }
             }
 
             await file.close(fs_account_config);
             file = null;
+            object_sdk.throw_if_aborted();
+
             // wait for the last drain if pending.
             if (drain_promise) {
                 await drain_promise;
                 drain_promise = null;
+                object_sdk.throw_if_aborted();
             }
 
             // end the stream
             res.end();
```

Comment:
I wonder - doesn't `res.end()` emit the 'close' even on the response? If so then would our code in object_sdk.setup_abort_controller that registers on the 'close' event might not work correctly and cause our next "throw_if_aborted" (the one after wait_finished) to fail? BTW notice that originally we did not register on the 'close' event of the response (in the original code in s3_get_object), and during our testing we decided to add it because we thought it was missing - perhaps it was not missing and now that we register on request 'aborted' we are covered for that case. Need to validate it. If this is not right, then I would suggest to remove that last throw_if_aborted, and then check if wait_finished completes fine before close is emitted.